src/transport/raw/TCP.cpp

```diff
@@ -271,77 +282,85 @@ CHIP_ERROR TCPBase::SendAfterConnect(const PeerAddress & addr, System::PacketBuf
 CHIP_ERROR TCPBase::ProcessReceivedBuffer(Inet::TCPEndPoint * endPoint, const PeerAddress & peerAddress,
                                           System::PacketBufferHandle buffer)
 {
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-    size_t consumed = 0;
+    ActiveConnectionState * state = FindActiveConnection(endPoint);
+    VerifyOrReturnError(state != nullptr, CHIP_ERROR_INTERNAL);
+
+    // Ack as we take ownership.
+    endPoint->AckReceive(buffer->TotalLength());
```

Comment:
Acknowledging before return, after the loop, allows application to have backpressure against the TCP socket. Suggest moving to the end.